app/notification.go

```diff
@@ -58,7 +58,7 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 	hereNotification := false
 	channelNotification := false
 	allNotification := false
-	updateMentionChans := []store.StoreChannel{}
+	updateMentionChans := []*model.AppError{}
```

Comment:
You are changing it from concurrent to serial. The whole point of have here the list of channels is to execute the queries concurrently. I would change it to a list of channels using the explicit chan type (because you are returning only `*model.AppError`, maybe we can go for `[]chan *model.AppError`). After that you should use the explicit async approach for the calls to the function, appending the channels there.